resources/assets/components/CampaignsTable.js

```diff
@@ -0,0 +1,163 @@
+import { get } from 'lodash';
+import gql from 'graphql-tag';
+import React, { useEffect } from 'react';
+import { useQuery } from '@apollo/react-hooks';
+
+import Empty from './Empty';
+import { updateQuery } from '../helpers';
+
+const CAMPAIGNS_QUERY = gql`
+  query CampaignsIndexQuery($isOpen: Boolean!, $cursor: String) {
+    campaigns: paginatedCampaigns(
+      isOpen: $isOpen
+      orderBy: "pending_count,desc"
+      after: $cursor
+      first: 15
+    ) {
+      edges {
+        cursor
+        node {
+          id
+          internalTitle
+          pendingCount
+          causes {
+            name
+          }
+        }
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+      }
+    }
+  }
+`;
+
+/**
+ * Filter the given campaigns by current search term.
+ *
+ * @param  {Object} data - GraphQL response
+ * @return {Object}
+ */
+const filterCampaigns = (data, filter) => {
+  const search = filter.toLowerCase();
+
+  if (!data) {
+    return [];
+  }
+
+  return data.campaigns.edges.filter(campaign => {
+    if (search === '') {
+      return true;
+    }
```

Comment:
Not sure if it makes enough of a difference -- but do you think if we just returned the `edges` array point-blank on top of this filter method (or only call `filterCampaigns` if there's a a `filter`), we'd avoid having to filter through the campaigns by default even when there's no `search`? (or is that such a cheap process that it doesn't matter 🤷‍♂ )

Reply:
Oh good call! It definitely can't hurt to avoid doing we don't have to. 😄

Reply:
Updated in bc832f4.